components/claim/components/email_to_ldap.tsx

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import classNames from 'classnames';
+
+import {AuthChangeResponse} from '@mattermost/types/users';
+
+import {emailToLdap} from 'actions/admin_actions.jsx';
+
+import {localizeMessage} from 'utils/utils';
+import {t} from 'utils/i18n';
+import {ClaimErrors} from 'utils/constants';
+
+import LoginMfa from 'components/login/login_mfa';
+import LocalizedInput from 'components/localized_input/localized_input';
+
+type Props = {
+    email: string | null;
+    siteName?: string;
+    ldapLoginFieldName?: string;
+}
+
+export type SubmitOptions = {
+    loginId: string;
+    password: string;
+    token?: string;
+    ldapIdParam?: string;
+    ldapPasswordParam?: string;
+}
+
+const EmailToLDAP = ({email, siteName, ldapLoginFieldName}: Props) => {
+    const emailPasswordInput = useRef<HTMLInputElement>(null);
+    const ldapIdInput = useRef<HTMLInputElement>(null);
+    const ldapPasswordInput = useRef<HTMLInputElement>(null);
+
+    const [password, setPassword] = React.useState('');
+    const [ldapId, setLdapId] = React.useState<string | null>(null);
+    const [ldapPassword, setLdapPassword] = React.useState<string | null>(null);
+    const [passwordError, setPasswordError] = React.useState('');
+    const [ldapError, setLdapError] = React.useState('');
+    const [ldapPasswordError, setLdapPasswordError] = React.useState('');
+    const [serverError, setServerError] = React.useState('');
+    const [showMfa, setShowMfa] = React.useState(false);
+
+    const preSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+
+        const password = emailPasswordInput.current?.value;
+        if (!password) {
+            setPasswordError(localizeMessage('claim.email_to_ldap.pwdError', 'Please enter your password.'));
+            return;
+        }
+
+        const ldapId = ldapIdInput.current?.value.trim();
+        if (!ldapId) {
+            setLdapError(localizeMessage('claim.email_to_ldap.ldapIdError', 'Please enter your AD/LDAP ID.'));
+            return;
+        }
+
+        const ldapPassword = ldapPasswordInput.current?.value;
+        if (!ldapPassword) {
+            setLdapPasswordError(localizeMessage('claim.email_to_ldap.ldapPasswordError', 'Please enter your AD/LDAP password.'));
```

Comment:
Maybe we can also get rid of all the different state variables and just have on for the error. That would really simplify that behavior as well

Reply:
I see your point @neallred. @michelengelen  As I cant test these changes on my local, I didnt want to change state variables for errors and kept the original solution as suggested by @neallred

Reply:
ok, having them actually makes sense in a way... although I don't see harm in having the UI show more than one error... but since the old component did that as well we can keep the behavior here. It is just not very good looking and IMO could possibly be improved, even with keeping the previous behavior. But that would be out of scope for this PR, so we can keep it as is for now. 👍🏼